### DISCUSSION TITLE
[General]Method modify ReflectionServiceDescripto#addMethod

### DISCUSSION THREAD

**Discussion by vio-lin** [apache/dubbo#14908] [General]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

this method provider a public method to add `MethodDescriptor` to `ServiceDescriptor`  
it change the filed **methods**, but another filed `descToMethods` is not be updated  
when method be added.

```
   public void addMethod(MethodDescriptor methodDescriptor) {
        methods.computeIfAbsent(methodDescriptor.getMethodName(), k -> new ArrayList<>(1))
                .add(methodDescriptor);

//        descToMethods.computeIfAbsent(methodDescriptor.getMethodName(), k -> new HashMap<>(1))
//             .put(methodDescriptor.getParamDesc(), methodDescriptor);
    }
```

it come to desearialization problem, when i  add additional method by java api `ServiceDescriptor.addMethod`.

```
org.apache.dubbo.rpc.protocol.dubbo.DecodeableRpcInvocation#drawPts
```
i think it is reasonable to update an other filed when add method should we？

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
Have you changed your service in runtime?

**Reply by vio-lin** [reply]
yes，we need add common method in below process for legacy code 。  

org.apache.dubbo.config.ServiceListener#exported

and return result with filter like  ‘org.apache.dubbo.rpc.filter.GenericFilter’.

**Reply by AlbumenJ** [reply]
This scenario need to use `re-export`. However, `re-export` isn't well-tested on Triple~ :) Would you like to fix it?

**Reply by vio-lin** [reply]
i am willing to,could you give me a list of some relate classes.

**Reply by AlbumenJ** [reply]
You can try un-export a service and then export a new service. Check the service works fine or not. You may need to check the logic in `TripleProtocol`
